src/reaper/wait.rs

```diff
@@ -186,5 +172,17 @@ cfg_if::cfg_if! {
                 }
             }
         }
+
+        /// Tell if we are able to use this backend.
+        pub(crate) fn available() -> bool {
+            // Create a Pidfd for the current process and see if it works.
+            let result = process::pidfd_open(
```

Comment:
does this open an FD without closing it? If yes, this should be fixed.

Reply:
This opens an `OwnedFd`, which closes itself on drop